engine/lib/luajit-ffi-gen/tests/test_impl_callbacks.rs

```diff
@@ -0,0 +1,399 @@
+use luajit_ffi_gen::luajit_ffi;
+
+mod helpers;
+use helpers::*;
+
+#[derive(Default)]
+pub struct CallbackTest {
+    val_primitives: Vec<f32>,
+    val_noncopyable: Vec<Data>,
+}
+
+// NOTE: remove 'lua_ffi' parameter to see generated Lua file. Do not commit it!!!
+#[luajit_ffi(lua_ffi = false)]
+impl CallbackTest {
+    pub fn nth_primitive<F: FnOnce(f32) -> ()>(&self, index: usize, callback: F) {
```

Comment:
Is it mandatory to specify `-> ()` or is it just for testing?

Reply:
Oh no it isnt! I was just being explicit, I'll make sure to mix and match a bit.